actionpack/lib/action_dispatch/journey/scanner.rb

```diff
@@ -7,64 +7,62 @@
 module ActionDispatch
   module Journey # :nodoc:
     class Scanner # :nodoc:
+      STATIC_TOKENS = Array.new(150)
+      STATIC_TOKENS[".".ord] = :DOT
+      STATIC_TOKENS["/".ord] = :SLASH
+      STATIC_TOKENS["(".ord] = :LPAREN
+      STATIC_TOKENS[")".ord] = :RPAREN
+      STATIC_TOKENS["|".ord] = :OR
+      STATIC_TOKENS[":".ord] = :SYMBOL
+      STATIC_TOKENS["*".ord] = :STAR
+      STATIC_TOKENS.freeze
```

Comment:
Spoke with @tenderlove about this, and we decided to use an array here because it is faster than needing to compute a hash key for every lookup.

Reply:
Not only that, we don't want to extract strings from the input when possible. Requesting the byte returns an integer, which doesn't allocate. If we have an integer, it makes more sense to do the lookup from an array than a hash.